table/tables/tables.go

```diff
@@ -749,6 +753,56 @@ func (t *TableCommon) AddRecord(sctx sessionctx.Context, r []types.Datum, opts .
 	return recordID, nil
 }
 
+func (t *TableCommon) checkTruncatedClusteredPrimaryKeyDuplication(sctx sessionctx.Context, pkDts []types.Datum,
+	tableCols []*table.Column, pkIdx *model.IndexInfo) error {
+	truncatedDts := make([]types.Datum, len(pkDts))
+	copy(truncatedDts, pkDts)
```

Comment:
Can we add another function to check whether the clustered index needs to truncate data? For each record, if we copy its datums, that's a memory heavy operation.

Reply:
Or we move this copy after check truncates in the following function.

Reply:
For the prefix clustered index like `primary key(a(2))`, it takes at least two such memory allocations:
1. Truncate the adding record
2. Truncate the target record in table

What's more, it also need to invoke `tables.IterRecord()` to search the target record in the table.

Is the performance overhead unacceptable?

Reply:
duplications can be checked by Get operation.

Reply:
We need to check the key's prefix duplication. Does Get operation provide this feature?

Reply:
If the primary key is a prefix, the physical key in the storage is prefix too.